Equilobe.DailyReport.SL/DataService.cs

```diff
@@ -482,7 +482,70 @@ public void SaveIpnLogError(long id, string error)
             }
         }
 
+        public List<SourceControlOptions> GetAllReposSourceControlOptions(long instanceId)
+        {
+            var sourceControlOptions = new List<SourceControlOptions>();
+
+            using (var db = new ReportsDb())
+            {
+                var policyStrings = GetInstancePolicyStrings(instanceId);
+
+                if (!policyStrings.Any())
+                    return sourceControlOptions;
+
+                foreach (var policyString in policyStrings)
+                {
+                    var advancedSettings = new AdvancedReportSettings();
+
+                    Deserialization.XmlDeserialize<AdvancedReportSettings>(policyString)
+                        .CopyPropertiesOnObjects(advancedSettings);
+
+                    sourceControlOptions.Add(advancedSettings.SourceControlOptions);
+                }
+            }
+
+            return sourceControlOptions;
+        }
+
+        public List<User> GetInstanceUsers(long instanceId)
+        {
+            var users = new List<User>();
+
+            using (var db = new ReportsDb())
+            {
+                var policyStrings = GetInstancePolicyStrings(instanceId);
+
+                if (!policyStrings.Any())
+                    return users;
+
+                foreach (var policyString in policyStrings)
+                {
+                    var advancedSettings = new AdvancedReportSettings();
+
+                    Deserialization.XmlDeserialize<AdvancedReportSettings>(policyString)
+                        .CopyPropertiesOnObjects(advancedSettings);
+
+                    users.AddRange(advancedSettings.UserOptions);
```

Comment:
Keep in mind that the same user can work on different projects, so this list may have duplicate users.